llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -7274,6 +7274,7 @@ static SDValue combineSelectToBinOp(SDNode *N, SelectionDAG &DAG,
     const APInt &FalseVal = FalseV->getAsAPIntVal();
     if (~TrueVal == FalseVal) {
       SDValue Neg = DAG.getNegative(CondV, DL, VT);
+      FalseV = DAG.getFreeze(FalseV);
```

Comment:
This freeze looks unnecessary. If x is poison so is ~x and they're constants here to boot.